File: client/src/Services/AdminServices.js
import api from "../../config/api"

const AdminServices = {

    getAdminDashboardData: async () => {
        try {
            const response = await api.get(`/api/admin/dashboard`, { withCredentials: true });
            // console.log("Dashboard Data Response:", response);
            if(response?.data?.success){
                return{success:true, data: response.data};
            }else{
                return{succsess:false, message: response.message};
            }
        } catch (error) {
            return{sucsess:false, message: error.message};
        }
    },


    // getSystemInformation: async () => {
    //     try {
    //         const response = await api.get(`/api/admin/uses`, { withCredentials: true });
    //         console.log("Dashboard Data Response:", response);
    //         if(response?.data?.success){
    //             return{success:true, data: response.data};
    //         }else{
    //             return{succsess:false, message: response.message};
    //         }
    //     } catch (error) {
    //         console.error("Error fetching system information:", error);
    //         return{sucsess:false, message: error.message};
    //     }
    // },

    getAllUsers:async()=>{ 
        try {
            let response = await api.get("/api/admin/users", { withCredentials: true });
            // console.log("All Users Response:", response);
            if(response?.data?.success){
                return{success:true, data: response.data};
            }else{
                return{succsess:false, message: response.message};
            }
        } catch (error) {
            return{sucsess:false, message: error.message};
        }
    },

    goastUser:async (mail) => {
        try {
            let response = await api.post("/api/admin/gosting", { mail: mail }, { withCredentials: true });
            // console.log("Impersonate User Response:", response);
            if(response?.data?.success){
                return{success:true, data: response.data};
            }else{
                return{succsess:false, message: response.message};
            }
        } catch (error) {
            return{sucsess:false, message: error.message};
        }
    },

    deleteUser:async (mail) => {
        try {
            let response = await api.delete(`/api/admin/delete/${mail}`, {withCredentials: true });
            console.log("Delete User Response:", response);
            if(response?.data?.success){
                return{success:true, data: response.data};
            }else{
                return{succsess:false, message: response.message};
            }
        }
        catch (error) {
            return{sucsess:false, message: error.message};
        }
    },

    getAllBots:async () => {
        try {
            let response = await api.get("/api/admin/bots", { withCredentials: true });
            console.log("All Bots Response:", response);
            if(response?.data?.success){
                return{success:true, data: response.data};
            }else{
                return{succsess:false, message: response.message};
            }
        } catch (error) {
            return{sucsess:false, message: error.message};
        }
    },

    deleteBot:async (botId) => {
        try {
            let response = await api.delete(`/api/admin/delete/bot/${botId}`, { withCredentials: true });
            console.log("Delete Bot Response:", response);
            if(response?.data?.success){
                return{success:true, data: response.data};
            }else{
                return{succsess:false, message: response.message};
            }
        } catch (error) {
            return{sucsess:false, message: error.message};
        }
    }

}


export default AdminServices;   

File: server/server/src/main/java/com/embediq/server/admin/services/AdminServices.java
package com.embediq.server.admin.services;

import com.embediq.server.bots.entity.Bot;
import com.embediq.server.bots.repo.BotRepo;
import com.embediq.server.chats.repo.ChatRepo;
import com.embediq.server.logs.LogsRepo;
import com.embediq.server.users.dto.LoginRepoce;
import com.embediq.server.users.entity.User;
import com.embediq.server.users.repo.UserRepo;
import com.embediq.server.utils.MyMailSender;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

@Service
public class AdminServices {

     @Autowired
    private UserRepo userRepo;

     @Autowired
    private BotRepo botRepo;

     @Autowired
    private ChatRepo chatRepo;

     @Autowired
     private LogsRepo logsRepo;

     @Autowired
     private MyMailSender sender;


     public Map<String,Object> getDashboardStats(){
           Map<String , Object> stats= new HashMap<>();
           stats.put("totalUser",userRepo.count());
           stats.put("activeBots",botRepo.countBotByIsBotActive(true));
           stats.put("totalChats",chatRepo.count());
         LocalDateTime tenMinsAgo = LocalDateTime.now().minusMinutes(10);
         stats.put("liveTraffic", logsRepo.countByCreatedAtAfter(tenMinsAgo));
         stats.put("recentActivity", logsRepo.findTop5ByOrderByCreatedAtDesc());
         return stats;
     }

    public List<LoginRepoce> getAllUsers(){
         List<User>  user =  userRepo.findAll();
        return user.stream().map(u -> {
            LoginRepoce response = new LoginRepoce();
            response.setEmail(u.getEmail());
            response.setRole(u.getRole().name());
            return response;
        }).collect(Collectors.toList());
    }

    public Boolean deleteUser(String mail){
         User user = userRepo.findByEmail(mail)
                 .orElseThrow(()->new RuntimeException("user not found"));
         userRepo.delete(user);
         return  true;
    }

    public List<Bot> getAllBots(){
         return  botRepo.findAll();
    }

    public Boolean deleteBot(String botId){
         Bot bot = botRepo.findById(botId)
                 .orElseThrow(()->new RuntimeException("bot not found"));
         botRepo.delete(bot);
        return  true;
    }
}
